pkg/dockerauth/multi.go

```diff
@@ -0,0 +1,32 @@
+package dockerauth
+
+import (
+	"fmt"
+
+	"github.com/fsouza/go-dockerclient"
+)
+
+type multiAuthProvider struct {
+	providers []AuthProvider
+}
+
+func NewMultiAuthProvider(providers ...AuthProvider) *multiAuthProvider {
+	return &multiAuthProvider{
+		providers: providers,
+	}
+}
+
+func (p *multiAuthProvider) GetAuthConfiguration(registry string) (docker.AuthConfiguration, error) {
+	for _, provider := range p.providers {
+		authConf, err := provider.GetAuthConfiguration(registry)
+		if err == nil {
```

Comment:
Hmm, I wonder if we should have an `ErrNoAuthConfiguration` that we can return from AuthProvider implementations, and only try to next one if the error returned is equal to that, so we can bubble up actual errors, like we do in [package auth](https://github.com/remind101/empire/blob/19c5350e31210926504ed9cadf69a7f024866c66/server/auth/auth.go#L103-L109).

We probably wouldn't want errors like `No ECR authorization data for <registry>` from the ECR provider to go unnoticed.